pkg/cmd/cli/cmd/volume.go

```diff
@@ -590,14 +678,68 @@ func (v *VolumeOptions) removeVolumeFromSpec(spec *kapi.PodSpec, info *resource.
 	return nil
 }
 
+func sourceAccessMode(readOnly bool) string {
+	if readOnly {
+		return " read-only"
+	}
+	return ""
+}
+
+func describePersistentVolumeClaim(claim *kapi.PersistentVolumeClaim) string {
+	if len(claim.Spec.VolumeName) == 0 {
+		// TODO: check for other dimensions of request - IOPs, etc
+		if val, ok := claim.Spec.Resources.Requests[kapi.ResourceStorage]; ok {
+			return fmt.Sprintf("waiting for %sB allocation", val.String())
+		}
+		return "waiting to allocate"
+	}
+	// TODO: check for other dimensions of capacity?
+	if val, ok := claim.Status.Capacity[kapi.ResourceStorage]; ok {
+		return fmt.Sprintf("allocated %sB", val.String())
+	}
+	return "allocated unknown size"
+}
+
+func describeVolumeSource(source *kapi.VolumeSource) string {
+	switch {
+	case source.AWSElasticBlockStore != nil:
+		return fmt.Sprintf("AWS EBS %s type=%s partition=%d%s", source.AWSElasticBlockStore.VolumeID, source.AWSElasticBlockStore.FSType, source.AWSElasticBlockStore.Partition, sourceAccessMode(source.AWSElasticBlockStore.ReadOnly))
+	case source.EmptyDir != nil:
+		return "empty directory"
+	case source.GCEPersistentDisk != nil:
+		return fmt.Sprintf("GCE PD %s type=%s partition=%d%s", source.GCEPersistentDisk.PDName, source.GCEPersistentDisk.FSType, source.GCEPersistentDisk.Partition, sourceAccessMode(source.GCEPersistentDisk.ReadOnly))
+	case source.GitRepo != nil:
+		if len(source.GitRepo.Revision) == 0 {
+			return fmt.Sprintf("Git repository %s", source.GitRepo.Repository)
+		}
+		return fmt.Sprintf("Git repository %s @ %s", source.GitRepo.Repository, source.GitRepo.Revision)
+	case source.Glusterfs != nil:
+		return fmt.Sprintf("GlusterFS %s:%s%s", source.Glusterfs.EndpointsName, source.Glusterfs.Path, sourceAccessMode(source.Glusterfs.ReadOnly))
+	case source.HostPath != nil:
+		return fmt.Sprintf("host path %s", source.HostPath.Path)
+	case source.ISCSI != nil:
+		return fmt.Sprintf("ISCSI %s target-portal=%s type=%s lun=%s%s", source.ISCSI.IQN, source.ISCSI.TargetPortal, source.ISCSI.FSType, source.ISCSI.Lun, sourceAccessMode(source.ISCSI.ReadOnly))
+	case source.NFS != nil:
+		return fmt.Sprintf("NFS %s:%s%s", source.NFS.Server, source.NFS.Path, sourceAccessMode(source.NFS.ReadOnly))
+	case source.PersistentVolumeClaim != nil:
+		return fmt.Sprintf("pvc/%s%s", source.PersistentVolumeClaim.ClaimName, sourceAccessMode(source.PersistentVolumeClaim.ReadOnly))
+	case source.RBD != nil:
+		return fmt.Sprintf("Ceph RBD %v type=%s image=%s pool=%s%s", source.RBD.CephMonitors, source.RBD.FSType, source.RBD.RBDImage, source.RBD.RBDPool, sourceAccessMode(source.RBD.ReadOnly))
+	case source.Secret != nil:
+		return fmt.Sprintf("secret/%s", source.Secret.SecretName)
+	default:
+		return "unknown"
+	}
+}
```

Comment:
Comments addressed

On Aug 11, 2015, at 4:54 PM, Ravi Sankar Penta <notifications@github.com>
wrote:

In pkg/cmd/cli/cmd/volume.go
<https://github.com/openshift/origin/pull/4055#discussion_r36797552>:

> +	case source.HostPath != nil:
> +		return fmt.Sprintf("host path %s", source.HostPath.Path)
> +	case source.ISCSI != nil:
> +		return fmt.Sprintf("ISCSI %s target-portal=%s type=%s lun=%s%s", source.ISCSI.IQN, source.ISCSI.TargetPortal, source.ISCSI.FSType, source.ISCSI.Lun, sourceAccessMode(source.ISCSI.ReadOnly))
> +	case source.NFS != nil:
> +		return fmt.Sprintf("NFS %s:%s%s", source.NFS.Server, source.NFS.Path, sourceAccessMode(source.NFS.ReadOnly))
> +	case source.PersistentVolumeClaim != nil:
> +		return fmt.Sprintf("pvc/%s%s", source.PersistentVolumeClaim.ClaimName, sourceAccessMode(source.PersistentVolumeClaim.ReadOnly))
> +	case source.RBD != nil:
> +		return fmt.Sprintf("Ceph RBD %v type=%s image=%s pool=%s%s", source.RBD.CephMonitors, source.RBD.FSType, source.RBD.RBDImage, source.RBD.RBDPool, sourceAccessMode(source.RBD.ReadOnly))
> +	case source.Secret != nil:
> +		return fmt.Sprintf("secret/%s", source.Secret.SecretName)
> +	default:
> +		return "unknown"
> +	}
> +}

Any addition of new volumesource in the upstream, needs an entry here. I
don't know how we can keep track of that easily.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4055/files#r36797552>.